File: tests/fixtures/source-windows/csharp/namespace-class.cs
using System.Collections.Generic;

namespace Windows
{
    /// <summary>Keep report entries in insertion order.</summary>
    public sealed class ReportBook
    {
        private readonly List<Entry> entries = [];

        public void Add(string title, string owner)
        {
            entries.Add(new Entry(title, owner));
        }

        public IReadOnlyList<string> Render()
        {
            return entries.ConvertAll(
                entry => $"{entry.Title} — {entry.Owner}");
        }

        public int Count => entries.Count;

        private sealed record Entry(string Title, string Owner);
    }
}

File: tests/fixtures/source-windows/php/namespace-class.php
<?php
namespace Windows;

final class ReportBook
{
    /** @var list<array{title: string, owner: ?string}> */
    private array $entries = [];

    public function add(string $title, ?string $owner): void
    {
        $this->entries[] = ['title' => $title, 'owner' => $owner];
    }

    /** @return list<string> */
    public function render(): array
    {
        return array_map(
            static fn (array $entry): string =>
                $entry['title'] . ' — ' . ($entry['owner'] ?? 'unassigned'),
            $this->entries,
        );
    }

    public function size(): int
    {
        return count($this->entries);
    }
}
